### ISSUE TITLE
Per spec, `refresh_token` is optional, but plugin fails without it

### ISSUE THREAD

**Issue by lionello** [daggerhart/openid-connect-generic#36]
This line of code prevents login without `refresh_token`:
https://github.com/daggerhart/openid-connect-generic/blob/b64fab27be1a66d065a060703173a540c4b4b845/includes/openid-connect-generic-client-wrapper.php#L138

Here's the line from the OAuth spec regarding `refresh_token`: https://tools.ietf.org/html/rfc6749

```
   refresh_token
         OPTIONAL.  The refresh token, which can be used to obtain new
         access tokens using the same authorization grant as described
         in Section 6.
```

The OpenID-Connect spec does not redefine this.

**Comment by pjeby** [comment]
Now that PR #52 has landed, this should no longer be an issue.  Absence of a refresh token will simply result in a login session that expires when the corresponding access token does.

**Comment by pjeby** [comment]
After doing some more research, it seems that there is actually another way to revalidate besides a refresh token; one can simply run the authentication flow again.  The [keycloak-proxy OIDC client](https://github.com/gambol99/keycloak-proxy/) does this if you don't enable a backend store or cookie store for refresh tokens: it sets the access token as a cookie, and simply redirects to the IdP whenever the token or cookie expires.

I'm not sure it's worth implementing that flow, though.  (Although the idea of ditching all of WP's crazy cookie and session management does have *some* appeal...  :grinning:  )

**Comment by daggerhart** [comment]
Going to close this as it should be resolved per #52.  If we decide to allow for re-authentication flow in the future that can be another issue.  Thanks!

**Comment by lionello** [comment]
Thanks!
